plugins/kotlin/src/main/java/org/vineflower/kotlin/struct/DefaultArgsMap.java

```diff
@@ -0,0 +1,101 @@
+package org.vineflower.kotlin.struct;
+
+import org.jetbrains.java.decompiler.main.DecompilerContext;
+import org.jetbrains.java.decompiler.main.rels.MethodWrapper;
+import org.jetbrains.java.decompiler.modules.decompiler.exps.*;
+import org.jetbrains.java.decompiler.modules.decompiler.flow.DirectGraph;
+import org.jetbrains.java.decompiler.modules.decompiler.flow.DirectNode;
+import org.jetbrains.java.decompiler.modules.decompiler.stats.IfStatement;
+import org.jetbrains.java.decompiler.modules.decompiler.stats.Statement;
+import org.jetbrains.java.decompiler.struct.attr.StructLocalVariableTableAttribute;
+import org.jetbrains.java.decompiler.util.TextBuffer;
+import org.vineflower.kotlin.KotlinOptions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DefaultArgsMap {
+  private final Map<KParameter, Exprent> map;
+
+  private DefaultArgsMap(Map<KParameter, Exprent> map) {
+    this.map = map;
+  }
+
+  public static DefaultArgsMap from(MethodWrapper defaults, MethodWrapper calling, KParameter[] params) {
+    if (defaults == null) {
+      return new DefaultArgsMap(Map.of());
+    }
+
+    Map<KParameter, Exprent> map = new HashMap<>();
+
+    DirectGraph graph = defaults.getOrBuildGraph();
```

Comment:
This piece of code is significantly nontrivial. I think you should add some comments explaining what's going on here.